crapssim/table.py

```diff
@@ -1,27 +1,46 @@
-import copy
-import typing
-
-from crapssim.dice import Dice
-
-from .bet import Bet, BetResult
-from .point import Point
-from .strategy import BetPassLine, Strategy
+import copy
+import typing
+from typing import Generator, Iterable, Literal, TypedDict
+
+from crapssim.dice import Dice
+
+from .bet import Bet, BetResult, DicePair
+import crapssim.bet as betmod
+from .point import Point
+from .strategy import BetPassLine, Strategy
 
 __all__ = ["TableUpdate", "TableSettings", "Table", "Player"]
 
 
-class TableUpdate:
-    """Object for processing a table after the dice has been rolled."""
-
-    def run(
-        self,
-        table: "Table",
-        dice_outcome: typing.Iterable[int] | None = None,
-        run_complete: bool = False,
-        verbose: bool = False,
-    ):
-        """Run through the roll logic of the table."""
-        self.run_strategies(table, run_complete, verbose)
+class TableUpdate:
+    """Helpers for progressing the table state after each roll."""
+
+    def run(
+        self,
+        table: "Table",
+        dice_outcome: DicePair | None = None,
+        run_complete: bool = False,
+        verbose: bool = False,
+    ) -> None:
+        """Execute the full roll/update lifecycle.
+
+        Args:
+            table: Active table instance being updated.
+            dice_outcome: Optional dice pair to use instead of rolling.
+            run_complete: If True, skip strategy updates that place/remove bets.
+            verbose: If True, print descriptive output for debugging.
+
+        Returns:
+            None: Always returns ``None``.
+        """
+        # --- Illegal Put guard ---
+        if table.point != "On":
+            for player in table.players:
+                illegal_puts = [b for b in list(player.bets) if isinstance(b, betmod.Put)]
+                for b in illegal_puts:
+                    player.bets.remove(b)
+        # --- end guard ---
```

Comment:
Can you help me understand why this is necessary? Put bets aren't allowed when the point is Off, and they should resolve on a 7-out. It seems like this logic would just slow down the table running unnecessarily